api/envoy/config/filter/accesslog/v2/accesslog.proto

```diff
@@ -188,3 +191,34 @@ message ResponseFlagFilter {
     ]
   }];
 }
+
+// Filters gRPC requests based on their response status. If a gRPC status is not provided, the
+// filter will infer the status from the HTTP status code.
+message GrpcStatusFilter {
+  enum Status {
+    OK = 0;
+    CANCELED = 1;
+    UNKNOWN = 2;
+    INVALID_ARGUMENT = 3;
+    DEADLINE_EXCEEDED = 4;
+    NOT_FOUND = 5;
+    ALREADY_EXISTS = 6;
+    PERMISSION_DENIED = 7;
+    RESOURCE_EXHAUSTED = 8;
+    FAILED_PRECONDITION = 9;
+    ABORTED = 10;
+    OUT_OF_RANGE = 11;
+    UNIMPLEMENTED = 12;
+    INTERNAL = 13;
+    UNAVAILABLE = 14;
+    DATA_LOSS = 15;
+    UNAUTHENTICATED = 16;
+  }
+
+  // Logs only responses that have any one of the gRPC statuses in this field.
+  repeated Status statuses = 1;
```

Comment:
I'm not too familiar with Protobuf validation, but for a list of repeated items wouldn't we have to do a by-items constraint like `[(validate.rules).repeated.items.enum.defined_only = true]`?

I actually tried both yours and mine, yours returns an error (expecting the item not to be repeated), and mine results in a SIGSEGV&mdash;probably worth reporting someplace ([protoc-gen-validate](https://github.com/lyft/protoc-gen-validate)?) if I can reproduce it with a small test case.

Any ideas?

Reply:
Can you try the following? (Notice the space after `.repeated`.)

```proto
repeated Status statuses = 1 [(validate.rules).repeated .items.enum.defined_only = true];
```

Reply:
It still generates a segfault, with or without the space between `repeated` and `.items`.

Reply:
A PR is out on PGV right now to fix this issue: https://github.com/lyft/protoc-gen-validate/pull/140.

Reply:
The PR has been merged, but since Envoy depends on the 0.0.12 release of PGV, I won't be able to do this until the relevant commit is included in a release. I'll hold off until 0.0.13 is released.

Reply:
Thanks for digging into this. @rodaine can you tag a new release of PGV?

Reply:
v0.0.13 is now available :)

Reply:
I've created PR #5783 to update PGV.

Reply:
Validation added in 6ba655e. Thanks @rodaine & @venilnoronha!